typify-impl/src/convert.rs

```diff
@@ -1132,6 +1132,51 @@ impl TypeSpace {
         Ok((TypeEntryDetails::Unit.into(), metadata))
     }
 
+    /// Determine whether a schema's property name validation constraints can be handled
+    fn can_handle_pattern_properties(validation: &ObjectValidation) -> bool {
+        if !validation.required.is_empty() {
+            return false;
+        }
+
+        if !validation.properties.is_empty() {
+            return false;
+        }
+
+        let all_pattern_property_schemas_same = validation
+            .pattern_properties
+            .values()
+            .next()
+            .map(|first| {
+                validation
+                    .pattern_properties
+                    .values()
+                    .all(|schema| schema == first)
+            })
+            .unwrap_or(true);
+
+        // Ensure we have at least one pattern property and all pattern property
+        // schemas are the same
+        if validation.pattern_properties.is_empty() || !all_pattern_property_schemas_same {
+            return false;
+        }
```

Comment:
Does this do the same thing? (written in the browser; might not compile)

```suggestion
        // Ensure we have at least one pattern property and all pattern property
        // schemas are the same.
        let Some(first) = validation.pattern_properties.values().next() else { return false; };
        if !validation.pattern_properties.values().all(|schema| schema == first) {
            return false;
        }
```

Reply:
Yes it does, much cleaner :) that'll teach me to carry around code without carrying around any critical thinking.